libs/services/designer-client-services/src/lib/base/function.ts

```diff
@@ -53,4 +54,119 @@ export class BaseFunctionService implements IFunctionService {
     });
     return keysResponse?.default ?? 'NotFound';
   }
+
+  public async fetchApiDefinitionUrl(functionAppId: string) {
+    const response = await this.options.httpClient.get<any>({
+      uri: `https://management.azure.com/${functionAppId}/config/web`,
+      queryParameters: { 'api-version': this.options.apiVersion },
+    });
+    if (!response?.properties?.apiDefinition?.url) {
+      throw new Error('ApiDefinitionUrl not found');
+    }
+    return response?.properties?.apiDefinition?.url ?? '';
```

Comment:
lets not default to empty string

Reply:
Should we throw an error if it's not present?